clang/lib/AST/ExprConstant.cpp

```diff
@@ -7095,6 +7096,40 @@ class APValueToBufferConverter {
     return true;
   }
 
+  bool visitVector(const APValue &Val, QualType Ty, CharUnits Offset) {
+    const auto *VT = Ty->castAs<VectorType>();
+    unsigned VectorLength = Val.getVectorLength();
+
+    if (VT->isExtVectorBoolType()) {
+      // Special handling for OpenCL bool vectors: we need to pack the vector
+      // of 1-bit unsigned integers into a single integer with the corresponding
+      // bits set, then write out the resulting integer.
+
+      CharUnits VecWidthBits = Info.Ctx.getTypeSizeInChars(VT) * 8;
+
+      APSInt Bits(VecWidthBits.getQuantity());
```

Comment:
```suggestion
      APSInt Bits(Info.Ctx.getTypeSize(VT));
```

You can just ask for the size in bits directly. (Also, please only use `CharUnits` for representing a number of `char` units, not bits.)

Reply:
Sorry, I pushed another commit right afterwards in which this is changed to not store bits. Should be correct now